libs/wingsdk/src/ex/dynamodb-table.ts

```diff
@@ -0,0 +1,396 @@
+ */
+export interface DynamodbTransactWriteItemsProps {
+  /**
+   * The write transact items.
+   */
+  readonly transactItems: DynamodbTransactWriteItem[];
+}
+
+/**
+ * Inflight interface for `DynamodbTable`.
+ */
+export interface IDynamodbTableClient {
+  /**
+   * Put an item into the table.
+   * @param item data to be inserted.
+   * @param props dynamodb PutItem props.
+   * @inflight
+   */
+  putItem(item: Json, props?: DynamodbTablePutItemProps): Promise<void>;
+
+  /**
+   * Get an item from the table.
+   * @param key key of the item.
+   * @param props dynamodb UpdateItem props.
+   * @inflight
+   */
+  updateItem(key: string, props?: DynamodbTableUpdateItemProps): Promise<Json>;
+
+  /**
+   * Delete an item from the table.
+   * @param key key of the item.
+   * @inflight
+   */
+  deleteItem(key: string): Promise<void>;
+
+  /**
+   * Get an item from the table.
+   * @param key key of the item.
+   * @inflight
+   */
+  getItem(key: string): Promise<Json>;
+
+  /**
+   * Get the table.
+   * @inflight
+   */
+  scan(): Promise<Array<Json>>;
+
+  /**
+   * Perform a synchronous write operation that groups up to 100 action requests.
+   * @param props properties for the transact write items operation.
+   * @inflight
+   */
+  transactWriteItems(props: DynamodbTransactWriteItemsProps): Promise<void>;
+}
+
+/**
+ * List of inflight operations available for `DynamodbTable`.
+ * @internal
+ */
+export enum DynamodbTableInflightMethods {
+  /** `DynamodbTable.putItem` */
+  PUT_ITEM = "putItem",
+  /** `DynamodbTable.updateItem` */
+  UPDATE_ITEM = "updateItem",
+  /** `DynamodbTable.deleteItem` */
+  DELETE_ITEM = "deleteItem",
+  /** `DynamodbTable.getItem` */
+  GET_ITEM = "getItem",
+  /** `DynamodbTable.scan` */
+  SCAN = "scan",
+  /** `DynamodbTable.transactWriteItems` */
+  TRANSACT_WRITE_ITEMS = "transactWriteItems",
+}
+
+/**
+ * Base class for `DynamodbTable` Client.
+ */
+export abstract class DynamodbTableClientBase implements IDynamodbTableClient {
+  /**
+   * @param tableName the table name.
+   * @param primaryKey the table primary key.
+   */
+  constructor(protected tableName: string, protected primaryKey: string) {}
+
+  /**
+   * Dynamodb table client.
+   * @internal
+   */
+  public abstract _rawClient(): Promise<DynamoDBClient>;
+
+  public async putItem(
+    item: Json,
+    props?: DynamodbTablePutItemProps
+  ): Promise<void> {
+    const client = await this._rawClient();
+    await client.send(
+      new PutItemCommand({
+        TableName: this.tableName,
+        Item: marshall(item),
+        ConditionExpression: props?.conditionExpression,
+      })
+    );
+  }
+
+  public async updateItem(
+    key: string,
+    props?: DynamodbTableUpdateItemProps
+  ): Promise<Json> {
+    const client = await this._rawClient();
+    const result = await client.send(
+      new UpdateItemCommand({
+        TableName: this.tableName,
+        Key: { [this.primaryKey]: { S: key } },
```

Comment:
If we're going to marshall automatically, maybe we should do it here as well:

```suggestion
        Key: marshall({ [this.primaryKey]: key }),
```